cloud/deploys/bin/azure/key-vault.sh

```diff
@@ -0,0 +1,61 @@
+#! /usr/bin/env bash
+
+#######################################
+# Create resource group
+# Arguments:
+#   1: The resource group name for the key vault
+#   2: The location of the key vault
+#######################################
+function key_vault::create_resource_group(){
+    if [ $(az group exists --name "${1}") = false ]; then
+        az group create --name "${1}" -l "${2}"
+    fi
+}
+
+#######################################
+# Create key vault
+# Arguments:
+#   1: The resource group name for the key vault
+#   2: The location of the key vault
+#   3: The name of the key vault 
+#######################################
+function key_vault::create_vault(){
+    
+    az keyvault create \
+        --name "${3}" \
+        --resource-group "${1}"\
+        --location "${2}" \
+        --enable-rbac-authorization
+}
+
+#######################################
+# Add a secret to the key vault
+# Arguments:
+#   1: The name of the key vault
+#   2: The name of the secret (i.e. "postgres-password")
+#   3: The value of the secret
+#######################################
+function key_vault::add_secret(){
+    az keyvault secret set \
+        --vault-name "${1}" \
+        --name "${2}" \
+        --value "${3}"
+}
+#######################################
+# Generates and adds secrets to the key vault
```

Comment:
suggestion (blocking): please expand on this description a little more e.g. "for each argument after the first, generates a random secret value and stores it in key vault using the argument as the secret name" or somesuch

Reply:
done